releaseNotes/Microsoft.Featuremanagement.md

```diff
@@ -4,6 +4,99 @@
 # Microsoft.FeatureManagement.AspNetCore
 [Source code ][source_code_web] | [Package (NuGet)][package_web] | [Samples][samples_web] | [Product documentation][docs]
 
+## 4.0.0-preview - January 4, 2024
+
+### Variants
+
+Variants provide the ability to surface different variations of a feature, allowing different segments of an audience to get different variations of a feature, all within a single feature flag. Variants also contain a value that can either reference another ConfigurationSection or be declared inline on the flag.
+
+```json
+{ 
+    "Name": "Big", 
+    "ConfigurationReference": "ShoppingCart:Big" 
+},  
+{ 
+    "Name": "Small", 
+    "ConfigurationValue": {
+        "Size": 300
+    }
+}
+```
+
+Variants are then assigned via Allocation, which is defined as a part of the feature flag. The fields are User, Group, Percentile, and a couple defaults.
+
+```json
+"Allocation": { 
+    "DefaultWhenEnabled": "Small", 
+    "DefaultWhenDisabled": "Small",  
+    "User": [ 
+        { 
+            "Variant": "Big", 
+            "Users": [ 
+                "Marsha" 
+            ] 
+        } 
+    ], 
+    "Group": [],
+    "Percentile": []
+}
+```
+
+A feature with Variants will still have a boolean result when called via `IsEnabledAsync`, following the usual filters. However, when a feature is called with `GetVariantAsync`, it will evaluate the allocation and return a `Variant` object. 
+
+```csharp
+Variant variant = await featureManager.GetVariantAsync(MyFeatureFlags.FeatureU);
+
+String variation = variant.Name;
+IConfigurationSection variantConfiguration = variant.Configuration;
+```
+
+For more details on Variants, see [here](https://github.com/microsoft/FeatureManagement-Dotnet/tree/release/v4?tab=readme-ov-file#variants).
+
+### Telemetry
+
+Telemetry offers the ability to emit events containing information about a feature evaluation. This can be used to ensure a flag is running as expected, or to see which users were given which features and why they were given the feature. To enable this functionality, two things need to be done.
```

Comment:
```suggestion
The feature management library now offers the ability to emit events containing information about a feature evaluation. This can be used to ensure a flag is running as expected, or to see which users were given which features and why they were given the feature. To enable this functionality, two things need to be done.
```